pkg/apis/proto/source/v1/udsource.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+option go_package = "github.com/numaproj/numaflow-go/pkg/apis/proto/source/v1";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+package source.v1;
+
+service UserDefinedSourceTransformer {
```

Comment:
Do not use `UserDefinedSourcexxxx`, as we will have real `UserDefinedSource` in the future (not transformer). Probably just `SourceTransformer`?